packages/assets-controllers/src/TokensController.ts

```diff
@@ -263,34 +270,55 @@ export class TokensController extends BaseController<
     });
   }
 
-  _instantiateNewEthersProvider(): any {
-    return new Web3Provider(this.config?.provider);
+  /**
+   * Get the relevant provider instance.
+   *
+   * @param networkClientId - Network Client ID.
+   * @returns Web3Provider instance.
+   */
+  #getProvider(networkClientId?: NetworkClientId): Web3Provider {
```

Comment:
do we need a method for this since we only use it one place?

Reply:
don't think it makes too much of a difference, but I'll move it